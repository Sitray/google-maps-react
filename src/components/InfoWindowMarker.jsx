import React, { useState } from "react";

import { InfoWindow, Marker } from "react-google-maps";

export const InfoWindowMarker = ({ position }) => {
  const [isOpen, setIsOpen] = useState(false);
  const handleToggle = () => {
    console.log("holaaa");
    isOpen ? setIsOpen(false) : setIsOpen(true);
    console.log(isOpen);
  };

  return (
    <div>
      <Marker position={position} onClick={handleToggle}>
        {isOpen && (
          <InfoWindow onCloseClick={handleToggle}>
            <p>hola</p>
          </InfoWindow>
        )}
      </Marker>
    </div>
  );
};
