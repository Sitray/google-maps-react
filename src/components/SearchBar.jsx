import React, { useState } from "react";
import { useDispatch } from "react-redux";

import PlacesAutoComplete, {
  geocodeByAddress,
  getLatLng,
} from "react-places-autocomplete";
import { markers } from "../actions/markers";

export const SearchBar = () => {
  const [address, setAddress] = useState("");

  const dispatch = useDispatch();

  const handleSelect = async (place) => {
    const result = await geocodeByAddress(place);
    const latAndLng = await getLatLng(result[0]);
    const getAaddress = result[0].formatted_address;
    setAddress(place);

    const { lat, lng } = latAndLng;

    dispatch(markers(getAaddress, lat, lng));
  };

  return (
    <div>
      <PlacesAutoComplete
        value={address}
        onChange={setAddress}
        onSelect={handleSelect}
      >
        {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => (
          <div>
            <input
              {...getInputProps({
                placeholder: "Busca la direcciÃ³n",
                required: true,
              })}
            />
            {loading ? <div>loading...</div> : null}

            {suggestions.map((suggestion, i) => {
              const style = {
                backgroundColor: suggestion.active ? "#2596be" : " #fff",
              };

              return (
                <div
                  key={i++}
                  {...getSuggestionItemProps(suggestion, { style })}
                >
                  {suggestion.description}{" "}
                </div>
              );
            })}
          </div>
        )}
      </PlacesAutoComplete>
    </div>
  );
};
