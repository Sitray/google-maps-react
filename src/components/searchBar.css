.searchbar {
  position: absolute;
  z-index: -1;
  align-self: center;
}
