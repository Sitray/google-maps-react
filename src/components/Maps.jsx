import React from "react";
import { withScriptjs, withGoogleMap, GoogleMap } from "react-google-maps";
import { useSelector } from "react-redux";

import { InfoWindowMarker } from "./InfoWindowMarker.jsx";

const Map = () => {
  const initialPosition = {
    lat: 41.40649793217181,
    lng: 2.174371444383683,
    InitialAddress: "Barcelona, Spain",
  };

  const state = useSelector((state) => state.marker);

  return (
    <div>
      <GoogleMap defaultCenter={initialPosition} defaultZoom={12}>
        {" "}
        {state.map((marker, i = 0) => {
          return (
            <InfoWindowMarker
              key={i++}
              position={{ lat: marker.latitude, lng: marker.longitude }}
              address={marker.address}
            />
          );
        })}
      </GoogleMap>
    </div>
  );
};

export default withScriptjs(withGoogleMap(Map));
