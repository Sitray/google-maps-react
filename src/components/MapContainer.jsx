import React, { Component } from "react";

import Map from "./Maps";
import { SearchBar } from "./SearchBar";

const mapUrl = `https://maps.googleapis.com/maps/api/js?v=3.exp&key=AIzaSyDMqBNkJ3WeEsG7rt__v7W1If9mUPcxkNo`;
export class MapContainer extends Component {
  render() {
    return (
      <div>
        <Map
          googleMapURL={mapUrl}
          containerElement={<div style={{ height: "400px" }} />}
          mapElement={<div style={{ height: "400px" }} />}
          loadingElement={<p>cargando</p>}
        />
        <div>
          <SearchBar />
        </div>
      </div>
    );
  }
}
