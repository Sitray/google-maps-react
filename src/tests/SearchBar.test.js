import React from "react";

import { shallow } from "enzyme";
import { SearchBar } from "../components/SearchBar";
import { Provider } from "react-redux";

import configureStore from "redux-mock-store";

describe("test on <SearchBar />", () => {
  const initialState = { id: 1 };
  const mockStore = configureStore();

  const wrapper = shallow(
    <Provider store={mockStore(initialState)}>
      <SearchBar />
    </Provider>
  );
  test("should display the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
