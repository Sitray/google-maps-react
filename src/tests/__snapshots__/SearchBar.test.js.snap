// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <SearchBar /> should display the component 1`] = `ShallowWrapper {}`;
